### PR TITLE
libassert: new recipe

### PR DESCRIPTION
**Pull request by Tsche**
Specify library name and version:  **libassert/1.1** and **libassert/1.0**

As mentioned in https://github.com/conan-io/conan-center-index/issues/4 I am developing a library that depends on libassert. After consulting with libassert's author I've decided to write a conan recipe for libassert. Looking forward to getting it merged :)

I have added version `1.0` as well since there was a namespacing change in `1.1`.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libassert/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - "https://github.com/jeremy-rifkin/libassert/archive/refs/tags/v1.1.tar.gz"
+    sha256: "01e1b99bf4dacd816fceeb5cc7efdc8875ed21295a62396584a0e7649b944c67"
+  "1.0":
+    url:
+      - "https://github.com/jeremy-rifkin/libassert/archive/refs/tags/v1.0.tar.gz"
+    sha256: "e1bb3b50767994ca4d0f60b7977b279cf32b8569ff92c5830e7a1de567b82fd5"
+
+patches:
+  "1.1":
+    - patch_file: "patches/1.1/0001-fix-max_line_number_width-miscalculation.patch"
```
Comment (RubenRBS) [inline recipes/libassert/all/conandata.yml:14]: Can you please add `patch_type` fields? It helps us a lot when reviewing patches to know their pourpose and origin!

**Review by RubenRBS — COMMENTED** [review]
Hi @Tsche thanks a lot for your contribution and patience while we process the rest of the PR review backlog

The PR looks good but I see you have some failures in the test package. Let me know if you need any help getting this thru the finish line :)

File: recipes/libassert/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, export_conandata_patches
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
```
Comment (RubenRBS) [inline recipes/libassert/all/conanfile.py:15]:
```suggestion
class LibAssertConan(ConanFile):
```

File: recipes/libassert/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    name = "libassert"
+    description = "The most over-engineered and overpowered C++ assertion library."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jeremy-rifkin/libassert"
```
Comment (RubenRBS) [inline recipes/libassert/all/conanfile.py:20]: Can you add a [`package_type`](https://docs.conan.io/2/reference/conanfile/attributes.html#package-type) here please?

**Comment by RubenRBS** [comment]
Hah! I just saw the library is by @jeremy-rifkin! 👋

File: recipes/libassert/all/conandata.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      patch_description: "Fixes Windows DLL CMake install"
+  "1.0":
+    - patch_file: "patches/1.0/0001-add-cstring-include.patch"
+      patch_type: "bugfix"
+      patch_description: "Add missing include"
+    - patch_file: "patches/1.0/0002-cmake-backport.patch"
+      patch_type: "portability"
+      patch_description: "Port back CMake install"
+    - patch_file: "patches/1.0/0003-fix-max_line_number_width-miscalculation.patch"
+      patch_type: "bugfix"
```
Comment (RubenRBS) [inline recipes/libassert/all/conandata.yml:31]: Quick question: are these patches official or custom made for Conan?

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Thanks a lot for your addressing the changes :)

File: recipes/libassert/all/conandata.yml (hunk 4 above)
Comment (Tsche) [reply recipes/libassert/all/conandata.yml:31]: The miscalculation bug is already merged back and will therefore be part of the next release, the other 1.1 patches will soon follow.

Other than that I'll have to check with Jeremy :)

**Comment by jeremy-rifkin** [comment]
The patches look good to me and I'll apply them to the upstream repo (A couple of them I think have already been merged)

**Review by RubenRBS — COMMENTED** [review]
@Tsche thanks a lot for your patience while we processes the review backlog and got to your PR. After the confirmation from Jeremy that the patches look good, this LGTM :)

**Review by RubenRBS — APPROVED** [review]
(actually approving the PR now)

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**